src/Confluent.Kafka/Producer.cs

```diff
@@ -191,11 +191,11 @@ public void HandleDeliveryReport(Message message)
             }
         }
 
-        private void Produce(
+        internal void Produce(
```

Comment:
again a bit ugly that this is exposed internally and accessed directly by `Produce<TKey, TValue>`. But having `DateTime` nullable leads to a more convenient implementation (and 1 less level of indirection).